### PR TITLE
add snapcraft packaging info

### LINKED ISSUE #1895: snapcraft build
any interest in having a snapcraft build like:
https://github.com/OGRECave/ogre-meshviewer/blob/master/snapcraft.yaml

it has the advantage that ror will be findable in the snap store and run sandboxed. One could even restrict it to its own folder so malicious content downloaded from the web can not do any harm.

### PR DESCRIPTION
**Pull request by paroj**
fixes #1895

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        source-tag: master
+        source-depth: 1
+        configflags:
+            - -DCMAKE_BUILD_TYPE=Release
+            - -DCMAKE_INSTALL_PREFIX=/usr/
+            - -DBUILD_REDIST_FOLDER=TRUE
+        override-build: |
+            snapcraftctl build
+            make zip_and_copy_resources
+            cp -R bin/content $SNAPCRAFT_PART_INSTALL/root/build_rigs-of-rods/parts/ror/build/redist/bin/
```
Comment (paroj) [inline snapcraft.yaml]: ideally ror should have a proper install target besides `BUILD_REDIST_FOLDER`, so no fixup is needed

**Comment by paroj** [comment]
bonus points for storing stuff in `$SNAP_USER_COMMON` instead of `~/.rigsofrods` if present:
https://docs.snapcraft.io/environment-variables/7983

**Comment by paroj** [comment]
mofilereader has to be fixed first https://github.com/AnotherFoxGuy/MofileReader/issues/2 it builds fine without for now.

**Comment by paroj** [comment]
fix networking part, added sound permission and mofilereader

**Comment by paroj** [comment]
https://snapcraft.io/rigs-of-rods

**Comment by ulteq** [comment]
@paroj, @AnotherFoxGuy can we merge this or do we have to fix the `BUILD_REDIST_FOLDER` thing first?

**Comment by paroj** [comment]
you can merge this. There is a workaround for BUILD_REDIST_FOLDER

**Comment by paroj** [comment]
FYI I keep rebasing this only partly to remind you to merge this - the main reason is that a commit triggers a rebuild & publish to the snap store.

**Pull request merged**